### PR TITLE
Db more routes combinators

### PR DESCRIPTION
**Pull request by danbornside**

### CODE REVIEW

File: lib/route/src/Obelisk/Route.hs [hunk 1]
```diff
@@ -859,4 +864,20 @@ renderObeliskRoute e r =
       enc = (pageNameEncoder . hoistParse (pure . runIdentity) e)
   in (T.pack . uncurry (<>)) $ encode enc r
 
+readShowEncoder :: (MonadError Text parse, Read a, Show a, Applicative check) => Encoder check parse a PageName
+readShowEncoder = singlePathSegmentEncoder . unsafeTshowEncoder
+
+
+integralEncoder :: (MonadError Text parse, Applicative check, Integral a) => Encoder check parse a Integer
+integralEncoder = prismEncoder (Numeric.Lens.integral)
```
Comment (3noch) [inline lib/route/src/Obelisk/Route.hs]: extra parens

**Comment by benkolera** [comment]
Was removing the deriveShowTagIdentity intentional? Because maybeEncoder requires a show instance, this breaks anything with a `(Maybe (R Subroute))`. Reflex-examples fails to compile with:

```
common/src/Common/Route.hs:71:58: error:
    • Could not deduce (Data.Dependent.Sum.ShowTag Example Identity)
        arising from a use of ‘maybeEncoder’
      from the context: b ~ Maybe (R Example)
        bound by a pattern with constructor:
                   FrontendRoute_Examples :: FrontendRoute (Maybe (R Example)),
                 in a case alternative
        at common/src/Common/Route.hs:71:7-28
```

**Comment by ali-abrar** [comment]
@benkolera Thanks for pointing this out. Didn't mean to bundle this with @danbornside's encoders.

**Comment by ali-abrar** [comment]
See https://github.com/obsidiansystems/obelisk/pull/365

**Pull request merged**